src/sxg_cbor.c

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include "libsxg/internal/sxg_cbor.h"
+
+#include <string.h>
+#include "libsxg/internal/sxg_buffer.h"
+
+static bool write_initial_bytes(uint8_t type_offset, uint64_t length,
```

Comment:
optional: CBOR spec itself allows small integers to be represented in larger types e.g. 0x17 in uint64_t. The rule "smallest type as possible" is by SXG's canonicity requirement for CBOR serialization. It'd be nice to leave some comments.

Reply:
OK.